tests/test_frame.py

```diff
@@ -518,71 +514,47 @@ def gethead(s: pd.Series, y: int) -> pd.Series:
         ),
         pd.DataFrame,
     )
+    check(
+        assert_type(df.apply(returns_dict, result_type="expand"), pd.DataFrame),
+        pd.DataFrame,
+    )
 
     # Check various return types for result_type="reduce" with default axis (0)
     check(
-        assert_type(
-            # Note that technically it does not make sense to pass a result_type of "reduce" to a scalar return
-            df.apply(returns_scalar, result_type="reduce"),
-            "pd.Series[int]",
-        ),
+        # Note that technically it does not make sense
+        # to pass a result_type of "reduce" to a scalar return
+        assert_type(df.apply(returns_scalar, result_type="reduce"), "pd.Series[int]"),
         pd.Series,
         int,
     )
     check(
-        assert_type(
-            # Note that technically it does not make sense to pass a result_type of "reduce" to a series return
-            df.apply(returns_series, result_type="reduce"),
-            pd.Series,
-        ),
+        # Note that technically it does not make sense
+        # to pass a result_type of "reduce" to a series return
+        assert_type(df.apply(returns_series, result_type="reduce"), pd.Series),
         pd.Series,  # This technically returns a pd.Series[pd.Series], but typing does not support that
     )
     check(
         assert_type(df.apply(returns_listlike_of_3, result_type="reduce"), pd.Series),
         pd.Series,
     )
-
-    # Check various return types for result_type="broadcast" with default axis (0)
     check(
-        assert_type(
-            # Note that technically it does not make sense to pass a result_type of "broadcast" to a scalar return
-            df.apply(returns_scalar, result_type="broadcast"),
-            pd.DataFrame,
-        ),
-        pd.DataFrame,
-    )
-    check(
-        assert_type(df.apply(returns_series, result_type="broadcast"), pd.DataFrame),
-        pd.DataFrame,
-    )
-    check(
-        assert_type(
-            # Can only broadcast a list-like of 2 elements, not 3, because there are 2 rows
-            df.apply(returns_listlike_of_2, result_type="broadcast"),
-            pd.DataFrame,
-        ),
-        pd.DataFrame,
```

Comment:
There assertions are moved to the bottom so that test cases for `result_type="broadcast"` are grouped together.